pkg/k8s/watchers/node.go

```diff
@@ -75,5 +79,8 @@ func (k *K8sWatcher) updateK8sNodeV1(oldK8sNode, newK8sNode *slim_corev1.Node) e
 	if err != nil {
 		return err
 	}
+
+	// Node change should be reflected in ciliumnode object as well # GH-15436
+	k.nodeDiscovery.UpdateCiliumNodeResource()
```

Comment:
Although this line fixes the bug, it won't be sufficient and it even might introduce other bugs. The `UpdateCiliumNodeResource` will only retry to update the CiliumNode after 10 retries. Ideally, this should be behind a controller that keeps retrying until it is successful.

One other issue is that `UpdateCiliumNodeResource` executes `mutateNodeResource` which performs an unnecessary GET into kube-apiserver and we have just received an event with the most up to date version of the K8s Node in the watcher.

Reply:
Just came around to this one, sorry for the delay.
Regarding the retries, isn't this a general problem of the update method anyway?
Currently during node discovery if the `CiliumNode` update fails 10 times, the it also will be incorrect / not complete.
Are you suggesting a rewrite of the Node Discovery procedure to act like a controller or just the IP syncing?

Regarding the unnecessary get: A transformation to a controller like behavior may also produce some unnecessary get requests. For instance the API-server might respond with a conflict which forces us to re-fetch the current `CiliumNode`. We might also need a [workqueue](https://pkg.go.dev/k8s.io/client-go/util/workqueue) to handle things like backoffs.

Reply:
As @gottwald says, this seems to be how the `CiliumNode` objects are updated everywhere else that needs to update them, so there are also other places where the `CiliumNode`s can become out-of-date.

To me, this feels like the correct solution to this particular issue, and it is the `UpdateCiliumNodeResource` method that needs refactoring to persist until it is successful? However I don't know the Cilium code so I might be missing something.

Reply:
However the refactoring of `UpdateCiliumNodeResource` could be a separate issue/PR. This PR certainly solves the issue I am seeing.

Reply:
The thing I would be more worried about is that a change to the node labels would also trigger a null-update of the corresponding `CiliumNode` resource.